chainer/links/connection/bilinear.py

```diff
@@ -60,32 +60,36 @@ def __init__(self, left_size, right_size, out_size, nobias=False,
         # This initialization is a modification of
         # that of Linear function.
 
+        shape = (left_size, right_size, out_size)
         if isinstance(initialW, (numpy.ndarray, cuda.ndarray)):
-            assert initialW.shape == self.W.shape
-        initializers.init_weight(self.W.data, initialW)
+            assert initialW.shape == shape
+        self.add_param('W', shape,
+                       initializer=initializers._get_initializer(initialW))
 
         if not self.nobias:
-            self.add_param('V1', (left_size, out_size))
-            self.add_param('V2', (right_size, out_size))
-            self.add_param('b', out_size)
-
+            V1_shape = (left_size, out_size)
+            V2_shape = (right_size, out_size)
+            b_shape = (out_size,)
             if isinstance(initial_bias, tuple):
                 V1, V2, b = initial_bias
+                if isinstance(V1, (numpy.ndarray, cuda.ndarray)):
+                    assert V1.shape == V1_shape
+                if isinstance(V2, (numpy.ndarray, cuda.ndarray)):
+                    assert V2.shape == V2_shape
+                if isinstance(b, (numpy.ndarray, cuda.ndarray)):
+                    assert b.shape == b_shape
+                V1 = initializers._get_initializer(V1)
+                V2 = initializers._get_initializer(V2)
+                b = initializers._get_initializer(b)
             elif initial_bias is None:
-                V1 = V2 = None
-                b = 0
+                V1 = V2 = initializers._get_initializer(None)
+                b = initializers.Constant(0)
             else:
                 raise ValueError('initial_bias must be tuple or None')
 
-            if isinstance(V1, (numpy.ndarray, cuda.ndarray)):
-                assert V1.shape == self.V1.shape
-            if isinstance(V2, (numpy.ndarray, cuda.ndarray)):
-                assert V2.shape == self.V2.shape
-            if isinstance(b, (numpy.ndarray, cuda.ndarray)):
-                assert b.shape == self.b.shape
-            initializers.init_weight(self.V1.data, V1)
-            initializers.init_weight(self.V2.data, V2)
-            initializers.init_weight(self.b.data, b)
+            self.add_param('V1', V1_shape, initializer=V1)
+            self.add_param('V2', V2_shape, initializer=V2)
+            self.add_param('b', b_shape, initializer=b)
```

Comment:
If you accept the above suggestion about variable notation, please fix this part from L:90~L:92 also.